File: tests/ingest/fixtures/python/shapes.py
"""Shapes — exercises classes, nested methods, internal + external imports,
a cross-file resolvable call, and an unresolvable attribute call."""

import math

from mathutils import square


class Circle:
    def __init__(self, radius):
        self.radius = radius

    def area(self):
        return math.pi * square(self.radius)  # cross-file call: -> mathutils.square


def describe(shape):
    return shape.area()  # attribute call on a parameter: unresolvable, recorded not guessed

File: tests/ingest/fixtures/typescript/shapes.ts
// Shapes — a class with nested methods, internal + external imports,
// a cross-file resolvable call, and an unresolvable attribute call.

import { square } from "./mathutils";

export class Circle {
  radius: number;

  constructor(radius: number) {
    this.radius = radius;
  }

  area(): number {
    return Math.PI * square(this.radius); // cross-file call: -> mathutils.square
  }
}

export function describe(shape: Circle): number {
  return shape.area(); // attribute call on a parameter: unresolvable, recorded not guessed
}
